### PR TITLE
Add generic base64 file field support

### PR DESCRIPTION
**Pull request by riklaunim**
I've refactored the Base64ImageField to create similar Base64FileField (and allow more customization for both fields).

### CODE REVIEW

File: drf_extra_fields/fields.py [hunk 1]
```diff
@@ -59,32 +53,63 @@ def to_internal_value(self, base64_data):
...
+            return super(Base64FieldMixin, self).to_internal_value(data)
         raise ValidationError(_('This is not an base64 string'))
 
     def get_file_extension(self, filename, decoded_file):
-        extension = imghdr.what(filename, decoded_file)
-        extension = "jpg" if extension == "jpeg" else extension
-        return extension
+        raise NotImplemented
 
     def to_representation(self, image):
```
Comment (yigitguler) [inline drf_extra_fields/fields.py]: Here we should change "image" to "file".

File: drf_extra_fields/fields.py [hunk 2]
```diff
@@ -59,32 +53,63 @@ def to_internal_value(self, base64_data):
...
+        extension = "jpg" if extension == "jpeg" else extension
+        return extension
+
+
+class Base64FileField(Base64FieldMixin, FileField):
+    """
+    A django-rest-framework field for handling file-uploads through raw post data.
+    It uses base64 for en-/decoding the contents of the file.
+    """
+    ALLOWED_TYPES = NotImplemented
```
Comment (yigitguler) [inline drf_extra_fields/fields.py]: We can add a helper message to the Not implemented error in order to provide better information to the programmer.

**Comment by yigitguler** [comment]
Hi @riklaunim,

Thank you for your contribution. We checked the PR with the team and apparently it is very well. We just noticed a name problem at one of the new methods. If you can fix that, we will be happy to merge the PR.

**Comment by riklaunim** [comment]
I'm working on it.

**Comment by yigitguler** [comment]
Thank you. We will prepare a new version shortly.

**Comment by yigitguler** [comment]
Could you please update the documentation as well?

**Pull request merged**